optimum/onnxruntime/modeling_ort.py

```diff
@@ -121,6 +131,12 @@ def __init__(self, model: ort.InferenceSession = None, config: PretrainedConfig
                 f" Use `ort_model.to()` to send the outputs to the wanted device."
             )
 
+        if "TensorrtExecutionProvider" in self.providers and self.use_io_binding:
+            logger.warning(
+                "There is no need to do IO binding for TensorrtExecutionProvider, `use_io_binding` will be set to False."
```

Comment:
Maybe basic, but you know why it is? Is it hurtful to use IOBinding with TensorrtExecutionProvider?

Reply:
Actually, there is no need to do IOBinding as TensorRT self-manages the memory. If you look into the source code of [TRT EP in ONNX Runtime](https://github.com/microsoft/onnxruntime/blob/2ecd1d662203d3fb5d77bf2aa97f6b8bcd7d1759/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L1778), you can see that they obtain the dimensions info of the inputs and outputs -> create buffers -> enqueue TRT context with created buffers -> run inference -> output put inside output buffer. There is no need to do the binding manually.